scripts/nix-profile.sh.in

```diff
@@ -9,9 +9,7 @@ if [ -n "$HOME" ] && [ -n "$USER" ]; then
         NIX_LINK_NEW="$HOME/.local/state/nix/profile"
     fi
     if [ -e "$NIX_LINK_NEW" ]; then
-        NIX_LINK="$NIX_LINK_NEW"
-    else
-        if [ -t 2 ] && [ -e "$NIX_LINK_NEW" ]; then
+        if [ -t 2 ] && [ -e "$NIX_LINK" ]; then
```

Comment:
turns out this is wrong. we should set NIX_LINK the same regardless of terminal-ness. need to split this up into one check for (old) nix link and another check for terminal, only for printing the message

Reply:
Marked as draft for now.

Reply:
rewritten to fix the logic